src/main/kotlin/com/atlassian/performance/tools/jirasoftwareactions/api/actions/WorkOnBacklog.kt

```diff
@@ -0,0 +1,110 @@
+import com.atlassian.performance.tools.jirasoftwareactions.api.WebJiraSoftware
+import com.atlassian.performance.tools.jirasoftwareactions.api.boards.ScrumBoard
+import com.atlassian.performance.tools.jirasoftwareactions.api.memories.AdaptiveBoardMemory
+import com.atlassian.performance.tools.jirasoftwareactions.api.memories.BoardMemory
+import com.atlassian.performance.tools.jirasoftwareactions.api.page.ViewBacklogPage
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import java.util.function.Predicate
+
+class WorkOnBacklog private constructor(
+    private val meter: ActionMeter,
+    private val editSprintChance: Float,
+    private val seededRandom: SeededRandom,
+    private val jiraSoftware: WebJiraSoftware,
+    private val filter: Predicate<ScrumBoard>,
+    private val boardMemory: BoardMemory<ScrumBoard>
+) : Action {
+    companion object {
+        @JvmField
+        val VIEW_BACKLOG = ActionType("View Backlog") { }
+
+        @JvmField
+        val EDIT_SPRINT = ActionType("Edit sprint") { }
+    }
+
+    private val logger: Logger = LogManager.getLogger(this::class.java)
+    private val newSprintName = "New Sprint Name"
+
+    override fun run() {
+        val board = boardMemory.recall { filter.test(it) }
+
+        if (board == null) {
+            logger.debug("Skipping View Backlog. I have no knowledge of Boards.")
+        } else {
+            val backlog = viewBacklog(board)
+            if (roll(editSprintChance)) {
+                editSprint(backlog)
+            }
+        }
+    }
+
+    private fun editSprint(backlog: ViewBacklogPage) {
+        val sprint = backlog.listSprints().firstOrNull()
+        if (sprint != null) {
+            meter.measure(
+                key = EDIT_SPRINT,
+                action = {
+                    sprint
+                        .openEditSprintDialog()
+                        .waitForDialog()
+                        .editSprintName(newSprintName)
+                        .validateSprintName(newSprintName)
+                }
+            )
+        }
+    }
+
+    private fun viewBacklog(board: ScrumBoard): ViewBacklogPage {
+        return meter.measure(
+            key = VIEW_BACKLOG,
+            action = {
+                val backlogBoard = jiraSoftware.goToBacklog(board.id)
+                backlogBoard.waitForBacklog()
+                backlogBoard
+            },
+            observation = { backlogBoard ->
+                val issueKeys = backlogBoard.getIssueKeys()
+                board.issuesInBacklog = issueKeys.size
+                IssuesOnBoard(issues = issueKeys.size).serialize()
+            }
+        )
+    }
+
+    private fun roll(
+        chance: Float
+    ): Boolean = (seededRandom.random.nextFloat() < chance)
+
+    class Builder(
+        private var jiraSoftware: WebJiraSoftware,
+        private var meter: ActionMeter
+    ) {
+        private var editSprintChance: Float = 0.5f
```

Comment:
When I see people working on backlog, they don't edit sprints 50% of the time. I don't have hard data from analytics, but intuitively its more like 10%.

Reply:
True, I will decrease the chance.